frontend/src/properties/LinksWebSocketContainer.tsx

```diff
@@ -0,0 +1,167 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { useSubscription } from '@apollo/client';
+import IconButton from '@material-ui/core/IconButton';
+import Snackbar from '@material-ui/core/Snackbar';
+import makeStyles from '@material-ui/core/styles/makeStyles';
+import Typography from '@material-ui/core/Typography';
+import CloseIcon from '@material-ui/icons/Close';
+import { useMachine } from '@xstate/react';
+import {
+  formRefreshedEventPayloadFragment,
+  subscribersUpdatedEventPayloadFragment,
+  widgetSubscriptionsUpdatedEventPayloadFragment,
+} from 'form/FormEventFragments';
+import { GQLLinksEventSubscription } from 'form/FormEventFragments.types';
+import gql from 'graphql-tag';
+import {
+  HandleCompleteEvent,
+  HandleSubscriptionResultEvent,
+  HideToastEvent,
+  LinksWebSocketContainerContext,
+  LinksWebSocketContainerEvent,
+  linksWebSocketContainerMachine,
+  SchemaValue,
+  ShowToastEvent,
+  SwitchSelectionEvent,
+} from 'properties/LinksWebSocketContainerMachine';
+import { Properties } from 'properties/Properties';
+import React, { useContext, useEffect } from 'react';
+import { RepresentationContext } from 'workbench/RepresentationContext';
+import { LinksWebSocketContainerProps } from './LinksWebSocketContainer.types';
+
+const linksEventSubscription = gql`
+  subscription linksEvent($input: LinksEventInput!) {
+    linksEvent(input: $input) {
+      __typename
+      ... on SubscribersUpdatedEventPayload {
+        ...subscribersUpdatedEventPayloadFragment
+      }
+      ... on WidgetSubscriptionsUpdatedEventPayload {
+        ...widgetSubscriptionsUpdatedEventPayloadFragment
+      }
+      ... on FormRefreshedEventPayload {
+        ...formRefreshedEventPayloadFragment
+      }
+    }
+  }
+  ${subscribersUpdatedEventPayloadFragment}
+  ${widgetSubscriptionsUpdatedEventPayloadFragment}
+  ${formRefreshedEventPayloadFragment}
+`;
+
+const usePropertiesWebSocketContainerStyles = makeStyles((theme) => ({
+  idle: {
+    padding: theme.spacing(1),
+  },
+}));
+
+/**
+ * Connect the Properties component to the GraphQL API over Web Socket.
+ */
+export const LinksWebSocketContainer = ({ editingContextId, selection }: LinksWebSocketContainerProps) => {
+  const classes = usePropertiesWebSocketContainerStyles();
```

Comment:
`useLinksWebSocketContainerStyles`, you may be pasting a bit too much code from the properties use case.